pkg/controller/controller.go

```diff
@@ -210,3 +216,30 @@ func (ctrl *CSIAttachController) syncPV() {
 	}
 	ctrl.handler.SyncNewOrUpdatedPersistentVolume(pv)
 }
+
+// shouldEnqueueVAChange checks if a changed VolumeAttachment should be enqueued.
+// It filters out changes in Status.Attach/DetachError - these were posted by the controller
+// just few moments ago. If they were enqueued, Attach()/Detach() would be called again,
+// breaking exponential backoff.
+func shouldEnqueueVAChange(old, new *storage.VolumeAttachment) bool {
+	if old.ResourceVersion == new.ResourceVersion {
+		// This is most probably periodic sync, enqueue it
+		return true
+	}
+	if new.Status.AttachError == nil && new.Status.DetachError == nil && old.Status.AttachError == nil && old.Status.DetachError == nil {
```

Comment:
Is this check needed or is it covered below?

Reply:
It is covered below, however, it saves us from DeepCopy and DeepEqual.